### PR TITLE
feat: Add commit split command

### PR DESCRIPTION
**Pull request by draftcode**

### CODE REVIEW

File: cmd/av/commit_split.go [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+	SilenceUsage: true,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		repo, err := getRepo()
+		if err != nil {
+			return err
+		}
+		if clean, err := repo.CheckCleanWorkdir(); err != nil {
+			return err
+		} else if !clean {
+			return errors.New("the workdir is not clean")
```
Comment (jainankit) [inline cmd/av/commit_split.go:25]: if this is a user facing message, you should explain it better
```suggestion
			return errors.New("The working directory is not clean, please stash or commit them before running split command")
```

Comment (jainankit) [reply cmd/av/commit_split.go:25]: Also what's the difference between `errors.Errorf` and `errors.New`?

File: cmd/av/commit_split.go [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+			return errors.Errorf("cannot update the branch %s to the new commit %s: %v", currentBranchName, newCommitOID, err)
+		}
+		// At this point, the HEAD is still a detached HEAD. Check out the branch.
+		// repo.CheckoutBranch errors out if the repository is at the detached head.
+		// We have to run git checkout in other ways.
+		if _, err := repo.Git("switch", currentBranchName); err != nil {
+			return errors.Errorf("cannot switch to the original branch: %v", err)
+		}
+
+		// TODO: We should rebase the stacks after split.
```
Comment (jainankit) [inline cmd/av/commit_split.go:109]: can you create an issue for this in GH?

File: internal/git/git.go [hunk 3]
```diff
@@ -156,6 +156,29 @@ func (r *Repo) CurrentBranchName() (string, error) {
 	return branch, nil
 }
 
+// CheckCleanWorkdir returns if the workdir is clean.
+func (r *Repo) CheckCleanWorkdir() (bool, error) {
+	out, err := r.Git("status", "--porcelain")
+	if err != nil {
+		return false, errors.Errorf("failed to check if the workdir is clean: %v", err)
```
Comment (jainankit) [inline internal/git/git.go:163]:
```suggestion
		return false, errors.Errorf("failed to check if the working directory is clean: %v", err)
```

**Review by jainankit — APPROVED** [review]

File: cmd/av/commit_split.go (hunk 1 above)
Comment (travigd) [reply cmd/av/commit_split.go:25]: See [`errors.Errorf`](https://pkg.go.dev/emperror.dev/errors#Errorf) and [`errors.New`](https://pkg.go.dev/emperror.dev/errors#New).

The tl;dr: is that `Errorf` allows using a format string while `New` only takes a single string argument.

File: cmd/av/commit_split.go [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+			colors.UserInput(commitOID), "\n",
+			"\n",
+			"In order to get back to ", colors.UserInput(commitOID), ",\n",
+			"you need to run ", colors.CliCmd("git switch --discard-changes --detach", commitOID), "\n",
+			colors.Failure("===================================================="), "\n",
+		)
+		return
+	}
+	_, _ = fmt.Fprint(
+		os.Stderr,
```
Comment (travigd) [inline cmd/av/commit_split.go:132]: I think `RECOVERY INSTRUCTION` is a bit too severe of a message here. Makes people think they're in a dangerous state and they could lose their work.

Maybe something like
```
====================================================
DETACHED HEAD

The commit split command aborted.
Your original branch {branch} was not modified.
Your Git repository is now in a detached HEAD state.
To revert your changes and switch to your original (unmodified) branch, run:

    git switch --discard-changes {branch}
====================================================

File: cmd/av/commit_split.go (hunk 2 above)
Comment (travigd) [reply cmd/av/commit_split.go:109]: At the very least, can you print out a message to the user that says
```
Run `av stack sync` to sync your stack if necessary.
```

**Review by travigd — APPROVED** [review]

File: cmd/av/commit_split.go (hunk 4 above)
Comment (draftcode) [reply cmd/av/commit_split.go:143]: Done.

File: internal/git/git.go (hunk 3 above)
Comment (draftcode) [reply internal/git/git.go]: Done.

File: cmd/av/commit_split.go (hunk 2 above)
Comment (draftcode) [reply cmd/av/commit_split.go:113]: Done.

File: cmd/av/commit_split.go (hunk 1 above)
Comment (draftcode) [reply cmd/av/commit_split.go]: Done.

**Pull request merged**